src/main/java/hudson/scm/IntegritySCM.java

```diff
@@ -899,24 +899,29 @@ private String getSource(EnvVars env) {
   {
     @Extension
     public static final DescriptorImpl INTEGRITY_DESCRIPTOR = new DescriptorImpl();
-    private ConnectionPoolDataSource dataSource;
+    private transient ConnectionPoolDataSource connectionPoolDataSource;
     private List<IntegrityConfigurable> configurations;
 
     public DescriptorImpl()
     {
       super(IntegritySCM.class, IntegrityWebUI.class);
       configurations = new ArrayList<IntegrityConfigurable>();
       load();
+    }
+    
+    @Override
+    public void load() {
+      super.load();

```

Comment:
Hard to say how this would have behaved in the past. `Descriptor.load` would have `unmarshal`ed over the fields set by the constructor. So some kind of hopelessly broken data source object would have been created, then immediately overwritten.

(https://github.com/jenkinsci/jenkins/pull/3196 is not directly relevant here, since the original code set this field _after_ calling `load`. If the order had been reversed, this plugin would I guess have starting throwing `NullPointerException`s as of 2.99.)